### DISCUSSION TITLE
Feature request: Add C# .NET

### DISCUSSION THREAD

**Discussion by mijenner** [trampgeek/jobe#76] [Ideas]
I'm teaching programming to high school students and we are urged to use as few languages as possible. Since the students have to use Arduino, which is C alike, we have choosen to go with C# as the OO expansion. 

It would be great if dotnet could be included, it should be cross platform for the console. 

Kind regards, 
Michael

**Comment by trampgeek** [comment]
Different users have different needs (different languages plus extra libraries like python3 numpy, matplotlib etc) so we don't attempt to cover all possibilities out of the box. But it's very easy to add C# to Jobe. Just login to the jobe server and type the command (as root)

sudo apt-get install mono-mcs

You may need to get an administrator to do this for you if you don't have direct access to the Jobe server. 

That makes C# available in the CodeRunner _multilanguage_ question type. That covers write-a-program questions where input is taken from standard input (albeit without echo to the screen). It's possible to write your own question types, such as write-a-function, write-a-class, specifically for C# but it does require a reasonable level of CodeRunner question-authoring expertise.
